client/my-sites/stats/controller.jsx

```diff
@@ -268,8 +265,9 @@ module.exports = {
 		} else if ( ! activeFilter || -1 === validModules.indexOf( context.params.module ) ) {
 			next();
 		} else {
-			if ( 'object' === typeof( site.options ) && 'undefined' !== typeof( site.options.gmt_offset ) ) {
-				momentSiteZone = i18n.moment().utcOffset( site.options.gmt_offset );
+			const gmt_offset = getSiteOption( context.store.getState(), siteId, 'gmt_offset' );
+			if ( gmt_offset || gmt_offset === 0 ) {
```

Comment:
This highlights an issue with using truthiness as a test for the value, since we awkwardly need to allow for the `0` (falsey) value by an or `||` check. What if instead we were to use a more targeted test like:

```js
'number' === typeof gmtOffset
```

Or:

```js
Number.isFinite( gmtOffset )
```